backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/util/Functions.java

```diff
@@ -0,0 +1,26 @@
+package ca.gov.dtsstn.cdcp.api.web.util;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonpatch.JsonPatch;
+import com.github.fge.jsonpatch.JsonPatchException;
+
+
+@Component
+public class Functions {
+
+	public <T> T applyJsonPatch(JsonPatch patch, T targetModel, Class<T> targetModelClass) {
+		try {
+			final ObjectMapper objectMapper = new ObjectMapper();
+			JsonNode patched = patch.apply(objectMapper.convertValue(targetModel, JsonNode.class));
+			return objectMapper.treeToValue(patched, targetModelClass);
```

Comment:
I'm no 100% sure, but it feels like there's a potential `NullPointerException` hiding in here somewhere. You should do some assertions on the incoming parameters so you can at least control the error message that is thrown if they are null.

Reply:
removed after #1701 was merged